src/clients/add/SamlConfig.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from "react";
+import { Controller, useFormContext } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import {
+  FormGroup,
+  Select,
+  SelectOption,
+  SelectVariant,
+  Switch,
+} from "@patternfly/react-core";
+
+import type { ClientForm } from "../ClientDetails";
+import { FormAccess } from "../../components/form-access/FormAccess";
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+
+export const Toggle = ({ name, label }: { name: string; label: string }) => {
+  const { t } = useTranslation("clients");
+  const { control } = useFormContext<ClientForm>();
+
+  return (
+    <FormGroup
+      hasNoPaddingTop
+      label={t(label)}
+      fieldId={label}
+      labelIcon={
+        <HelpItem
+          helpText={t(`clients-help:${label}`)}
+          forLabel={t(label)}
+          forID={label}
+        />
+      }
+    >
+      <Controller
+        name={name}
+        defaultValue="false"
+        control={control}
+        render={({ onChange, value }) => (
+          <Switch
+            id={name!}
+            data-testid={label}
+            label={t("common:on")}
+            labelOff={t("common:off")}
+            isChecked={value === "true"}
+            onChange={(value) => onChange("" + value)}
```

Comment:
A boolean primitive also has a `toString()` method that can be used instead.
```suggestion
            onChange={(value) => onChange(value.toString())}
```